src/App.tsx

```diff
@@ -1,14 +1,88 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { ShowTodos } from './types/ShowTodos';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodos, setSelectedTodos] = useState<ShowTodos>(ShowTodos.All);
+  const [query, setQuery] = useState<null | string>(null);
+  const [selectedTodoID, setSelectedTodoID] = useState<null | number>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(fetchedTodos => {
+        setTodos(fetchedTodos);
+        setIsLoading(false);
+      });
+  }, []);
+
+  const handleFilterSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    switch (event.target.value) {
+      case 'active':
+        setSelectedTodos(ShowTodos.Active);
+        break;
+
+      case 'completed':
+        setSelectedTodos(ShowTodos.Completed);
+        break;
+
+      default:
+        setSelectedTodos(ShowTodos.All);
+        break;
+    }
+  };
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  const clearQuery = () => setQuery(null);
+
+  const setTodoID = (todoID: number) => setSelectedTodoID(todoID);
```

Comment:
Do you really need a wrapper for the setSelectedTodoID here? It seems that this function just calls the `setSelectedTodoID`. You can call it directly

Reply:
Done